### PR TITLE
fix: Added in support for the timezone.location in boundaries package

### PR DESCRIPTION
**Pull request by Westonside**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### Updates
Updates in regards to #4930 

- Updated functions in the boundaries package to include support for `timezone.location()`
- Changed test names

### CODE REVIEW

File: stdlib/date/boundaries/boundaries.flux [hunk 1]
```diff
@@ -387,7 +391,15 @@ sunday = () => {
...
-    as = _timezone_convert(s: s)
+
+    // if the timezone offset is not 0 then timezone convert else truncate
+    as =
+        if int(v: location.offset) != int(v: 0ns) then
+            _timezone_convert(s: s)
+        else
+            date.truncate(t: s, unit: 1d)
+
+    // as = _timezone_convert(s: s)
```
Comment (onelson) [inline stdlib/date/boundaries/boundaries.flux:402]: Better to remove unwanted lines than just comment them out.

File: stdlib/date/boundaries/tests/boundaries_test.flux [hunk 2]
```diff
@@ -257,7 +257,21 @@ testcase thursday_test_tzt_timeable {
     testing.diff(want: want, got: got)
 }
 
-testcase friday_test_timeable {
+// Africa/Algiers
+testcase thursday_test_tzts_timeableasdffff {
```
Comment (onelson) [inline stdlib/date/boundaries/tests/boundaries_test.flux:261]: This test could probably use a better name.

File: stdlib/date/boundaries/boundaries.flux [hunk 3]
```diff
@@ -387,7 +391,15 @@ sunday = () => {
...
 month = (month_offset=0) => {
     s = date.truncate(t: today(), unit: 1mo)
-    as = _timezone_convert(s: s)
+
+    // if the timezone offset is not 0 then timezone convert else truncate
+    as =
+        if int(v: location.offset) != int(v: 0ns) then
+            _timezone_convert(s: s)
+        else
+            date.truncate(t: s, unit: 1d)
```
[the remark this answers is not in the document]
Comment (Westonside) [reply stdlib/date/boundaries/boundaries.flux:400]: You are right, it works without. Will remove

**Comment by Westonside** [comment]
I added a temporary fix to the date.go file because the offset set by `timezone.offset` was being ignored by the truncate function.

File: stdlib/date/date.go [hunk 4]
```diff
@@ -260,11 +260,14 @@ func init() {
 				if err != nil {
 					return nil, err
 				}
-				location, _, err := date.GetLocationFromObjArgs(args)
+				//was originally ignoring the offset _ so looked like location ,_,err Addressed in #5013
```
Comment (onelson) [inline stdlib/date/date.go:263]:
```suggestion
```

File: stdlib/date/date.go [hunk 5]
```diff
@@ -260,11 +260,14 @@ func init() {
...
+				location, offset, err := date.GetLocationFromObjArgs(args)
 				if err != nil {
 					return nil, err
 				}
 				intervalLocation, err := interval.LoadLocation(location)
+				//temporary fix change the location address issue #5013
```
Comment (onelson) [inline stdlib/date/date.go:269]:
```suggestion
				// TODO: offset previously ignored. Is this enough of a fix?
				//  Need to check to see if setting it here is a sufficient fix. Follow-up in #5013
```

**Review by onelson — APPROVED** [review]
Offered a small tweak to the comments re: #5013. Looks good otherwise.

**Pull request merged**